src/libstd/collections/hash/table.rs

```diff
@@ -126,9 +126,10 @@ unsafe impl<K: Send, V: Send> Send for RawTable<K, V> {}
 unsafe impl<K: Sync, V: Sync> Sync for RawTable<K, V> {}
 
 pub struct RawBucket<K, V> {
-    hash: *mut HashUint,
+    _hash: *mut HashUint,
     // We use *const to ensure covariance with respect to K and V
-    pair: *const (K, V),
+    _pair: *const (K, V),
+    idx: usize,
```

Comment:
If this is always within `[0..table_capacity)`, could you add a comment mentioning that.